databases/week3/sharing_meal_queries.sql

```diff
@@ -0,0 +1,87 @@
+USE sharing_meal;
+
+--meal
+SELECT title
+FROM meal;
+insert into meal (title, description, location, when_event, max_reservations, price, created_date) values ('Laziza Kheer', 'A sweet dish made with rice milk cream and nuts', 'Kastrup', '2022-02-11 09:16:50', 5, 89.99, '2021-12-02');
+SELECT *
+FROM meal
+WHERE id = 6;
+UPDATE meal
+SET title = 'Mutton Korma'
+WHERE id = 4;
+DELETE FROM meal
+WHERE id = 1;
+
+--reservation
+SELECT *
+FROM reservation;
+insert into reservation (number_of_guests, meal_id, created_date, contact_name, contact_phonenumber, contact_email) values ('1', '3', '2021-11-04', 'Sam Billings', '54108124', 'billings007@yahoo.com');
+SELECT *
+FROM reservation
+WHERE id = 8;
+UPDATE reservation
+SET number_of_guests = 4
+WHERE id = 8;
+DELETE FROM reservation
+WHERE id = 3;
+
+--review
+SELECT *
+FROM review;
+insert into review (title, description, stars, created_date, meal_id) values ('Great Service', 'The service was quick food taste is just fine', '4', '2021-11-10', 3);
+SELECT *
+FROM review
+WHERE id = 9;
+UPDATE review
+SET created_date = '2022-01-09'
+WHERE id = 9;
+
+DELETE FROM review 
+WHERE id = 11;
+--Get meals that has a price smaller than a specific price
+SELECT *
+FROM meal
+WHERE price < 150;
+--Get meals that still has available reservations
+SELECT *
```

Comment:
This is the trap query. You need to sum the number_of_guests over all reservations for a given meal and then compare that number with the max_reservations of that meal.

There are at least two ways of doing this, both involve stuff that we didnt talk about in class:
 - use the COALESCE() function: https://dev.mysql.com/doc/refman/8.0/en/comparison-operators.html#function_coalesce
 - use GROUP BY, SUM, and `HAVING: https://www.w3schools.com/sql/sql_having.asp

Also remember that there might be meals that don't have any reservations yet.